src/Compilers/Core/Portable/AdditionalTextFile.cs

```diff
@@ -43,18 +43,20 @@ public AdditionalTextFile(CommandLineSourceFile sourceFile, CommonCompiler compi
         /// Returns a <see cref="SourceText"/> with the contents of this file, or <c>null</c> if
         /// there were errors reading the file.
         /// </summary>
-        public override SourceText GetText(CancellationToken cancellationToken = default)
+        public override SourceText? GetText(CancellationToken cancellationToken = default)
         {
-            lock (_lockObject)
+            if (_text == null && _diagnostics == null)
```

Comment:
what's the current status of double-checked locking.  Is it guaranteed to be safe on .net fx/core? Does it work with these complex check patterns?

Reply:
can we just use Lazy (or LazyInitializer) instead?  I trust the framework far more here than any sort of manual double-checked lock.